hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -153,29 +153,29 @@ private void enrichHostHeader(Event event) {
 
   private Optional<String> getSanitizedHostValue(String value) {
     if (StringUtils.isNotBlank(value)) {
-      String host = sanitizeHostValue(value);
-      if (!LOCALHOST.equalsIgnoreCase(host)) {
-        return Optional.of(host);
+      Optional<String> host = sanitizeHostValue(value);
+      if (host.isPresent() && !LOCALHOST.equalsIgnoreCase(host.get())) {
```

Comment:
Do we need conditon - `StringUtils.isEmpty(host.get())` - for case where host is not `null` but an empty `""` string?

Reply:
`equalsIgnoreCase()` will take care of those.